torch/_dynamo/variables/ctx_manager.py

```diff
@@ -78,30 +81,42 @@ def enter(self, tx):
         options["source"] = (
             None if self.source is None else AttrSource(self.source, "__enter__")
         )
-        return variables.UserMethodVariable(
-            self.cm_obj.__enter__.__func__,
-            variables.UserDefinedObjectVariable(self.cm_obj, **options),
-            **options,
-        ).call_function(tx, [], {})
+        try:
+            return variables.UserMethodVariable(
+                self.cm_obj.__enter__.__func__,
+                variables.UserDefinedObjectVariable(self.cm_obj, **options),
+                **options,
+            ).call_function(tx, [], {})
+        except Exception as e:
+            log.exception(e)
+            unimplemented(
+                f"Context manager `{self.cm_obj}`'s `__enter__` method is not supported"
+            )
 
     def exit(self, tx, *args):
         options = VariableTracker.propagate(self)
         options["source"] = (
             None if self.source is None else AttrSource(self.source, "__exit__")
         )
-        x = variables.UserMethodVariable(
-            self.cm_obj.__exit__.__func__,
-            variables.UserDefinedObjectVariable(self.cm_obj, **options),
-            **options,
-        ).call_function(
-            tx,
-            [
-                variables.ConstantVariable(None),
-                variables.ConstantVariable(None),
-                variables.ConstantVariable(None),
-            ],
-            {},
-        )
+        try:
+            x = variables.UserMethodVariable(
+                self.cm_obj.__exit__.__func__,
+                variables.UserDefinedObjectVariable(self.cm_obj, **options),
+                **options,
+            ).call_function(
+                tx,
+                [
+                    variables.ConstantVariable(None),
+                    variables.ConstantVariable(None),
+                    variables.ConstantVariable(None),
+                ],
+                {},
+            )
+        except Exception as e:
```

Comment:
Yikes, does this need to a bare exception? Can it at least be a runtime exception?